sdk/core/core-amqp/src/requestResponseLink.ts

```diff
@@ -273,3 +228,75 @@ export class RequestResponseLink implements ReqResLink {
     return new RequestResponseLink(session, sender, receiver);
   }
 }
+
+type NormalizedInfo = {
+  statusCode: number;
+  statusDescription: string;
+  errorCondition: string;
+};
+
+// Handle different variations of property names in responses emitted by EventHubs and ServiceBus.
+const getCodeDescriptionAndError = (props: any): NormalizedInfo => {
+  if (!props) props = {};
+  return {
+    statusCode: (props[Constants.statusCode] || props.statusCode) as number,
+    statusDescription: (props[Constants.statusDescription] || props.statusDescription) as string,
+    errorCondition: (props[Constants.errorCondition] || props.errorCondition) as string
+  };
+};
+
+/**
+ * This is used as the onMessage handler for the "message" event on the receiver.
+ *
+ * (This is inspired from the message settlement sequence in service-bus SDK which
+ * relies on a single listener for settled event for all the messages.)
+ * The sequence is as follows:
+ * 1. User calls `await RequestResponseLink.sendRequest()`
+ * 2. This creates a `Promise` that gets stored in the _responsesMap
+ * 3. When the service acknowledges the response, this method gets called for that request.
+ * 4. We resolve() the promise from the _responsesMap with the message.
+ * 5. User's code after the sendRequest continues.
+ *
+ * @internal
+ * @ignore
+ */
+export function onMessageReceived(
+  context: EventContext,
+  connection: Connection,
+  responsesMap: Map<string, DeferredPromiseWithCallback>
+): void {
+  const message = context.message;
+  if (message) {
+    const responseCorrelationId = message.correlation_id;
+    if (responsesMap.has(responseCorrelationId as string)) {
+      const promise = responsesMap.get(
+        responseCorrelationId as string
+      ) as DeferredPromiseWithCallback;
```

Comment:
Is this cast needed?

Reply:
![image](https://user-images.githubusercontent.com/10452642/96053965-74c42800-0e35-11eb-978b-ba5349dc14f4.png)
`.get` either returns `DeferredPromiseWithCallback` or `undefined`, though we never set undefined.